### PR TITLE
[TypeDeclaration] Skip mixed union return on ReturnTypeDeclarationRector

### PR DESCRIPTION
**Pull request by samsonasik**
It currently generate invalid return type: 

```diff
-    public function get($id) {
+    public function get($id): \stdClass {
```

for code: 

```php
    /**
     * @return stdClass|array
     */
    public function get($id) {
        if (rand(0,1)) {
            return new stdClass();
        }

        return [];
    }
```

### CODE REVIEW

**Comment by samsonasik** [comment]
Updated fixture, it seems happen for union return.

**Comment by samsonasik** [comment]
The issue seems on `PHPStanStaticTypeMapper::mapToPhpParserNode()` which returns `Rector\PHPStanStaticTypeMapper\TypeMapper\ObjectTypeMapper` which should be `PhpParser\Node\UnionType`.

https://github.com/rectorphp/rector-src/blob/36ac7ab2843908b6f99ef67d372d58e4d5d22c13/packages/PHPStanStaticTypeMapper/PHPStanStaticTypeMapper.php#L39-L50

**Comment by samsonasik** [comment]
@TomasVotruba it also seems invalid when using can have return or not, like this:

```php
    public function get($id) 
    {
        if (rand(0, 1)) {
            return [];
        }
        
        // do nothing
    }
```

make:

```diff
-    public function get($id) {
+    public function get($id): void {
         if (rand(0, 1)) {
```

ref https://3v4l.org/irYVr

**Comment by samsonasik** [comment]
Fixed 🎉

File: packages/ReadWrite/NodeFinder/NodeUsageFinder.php [hunk 1]
```diff
@@ -25,7 +25,7 @@ public function __construct(
 
     /**
      * @param Node[] $nodes
-     * @return Variable[]
+     * @return never[]|Node[]
```
Comment (samsonasik) [inline packages/ReadWrite/NodeFinder/NodeUsageFinder.php]: @TomasVotruba handle UnionType https://github.com/rectorphp/rector-src/pull/393/commits/273dc5d09243b56449a889c163c775c602487ce5#diff-e21a1c8f1784f36751a42d937959b979ca4fed5ddf8f05ad74c8c4e72dc84dc3R55 seems make addition of `never[]` on various places.

Comment (samsonasik) [reply packages/ReadWrite/NodeFinder/NodeUsageFinder.php]: handled at https://github.com/rectorphp/rector-src/pull/393/commits/b17ef4f966bb5228ef3f2b74025c284bc3681cf3

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by samsonasik** [comment]
also added fixture use case for union mixed array never type on `AddArrayReturnDocTypeRector` 🎉

**Comment by TomasVotruba** [comment]
Thank you :+1: :clap:

**Pull request merged**